CharFeatures.py

```diff
@@ -0,0 +1,98 @@
+"""
+Lena Heide
+Matr.-Nr.: 767488
+30.4.2021
+"""
+
+from nltk.tokenize import RegexpTokenizer
+from AbstractFeatures import Features
+
+class stChFeats(Features):
+	"""
+	A class for extraction of character features from a list of strings.
+	Attributes
+	----------
+	text
+		A tokenized given text. An inherited attribute from an abstract
+		class Features().
+	Methods
+	-------
+	outputter
+		Count the number of uppercase, lowercase and numeric characters
+		and return an appropriate output.
+		An inherited method from an abstract class Features().
+    """
```

Comment:
Um ganz deutlich zu machen, inwiefern du die Original __init__ erweitert hast, könntest du auch folgendes tun:
```
    def __init__(self, text: list):
        super().__init__(text)
        # stores the length of the original strings for normalization
        self.nr_char = []
        for string in text:
            self.nr_char.append(len(string))
```
Was diese Variante einer __init__ macht, ist dass sie sämtlichen Code ausführt, der in der __init__ der Mutterklasse (also der abstrakten Klasse) steht. Also zB die Instanzvariable self.text initialisiert. Anschließend kann man für die Kindklasse noch zusätzliche Instanzvairablen wie hier self.nr_char definieren.